src/comparator.ml

```diff
@@ -0,0 +1,8 @@
+
+type 'a t = 'a -> 'a -> TotalOrder.t
+
+let make (compare : 'a -> 'a -> TotalOrder.t) (this : 'a) (that : 'a): TotalOrder.t =
```

Comment:
fromInt is less convenient. make is intended to return a (Comparator.t 'a) although I don't think I even expose it in the public api.